### PR TITLE
PSQLADM-396 proxy sql jenkins job is failing on centos 8

### PR DESCRIPTION
**Pull request by adivinho**
Screenshot 2022-06-15 at 18 20 41

### CODE REVIEW

File: proxysql/build-binary-proxysql [hunk 1]
```diff
@@ -12,7 +12,17 @@ set -o xtrace
...
-        sudo yum -y install cmake python2 gnutls-devel libtool || true
+        sudo yum -y install python2 gnutls-devel libtool || true
+        sudo yum remove -y cmake
+        sudo wget https://github.com/Kitware/CMake/releases/download/v3.19.8/cmake-3.19.8.tar.gz
+        sudo tar -zxvf cmake-3.19.8.tar.gz
+        cd cmake-3.19.8/
+        sudo ./bootstrap
+        sudo make
+        sudo make install
+        PATH=$PATH:/usr/local/bin
```
Comment (mohitj1988) [inline proxysql/build-binary-proxysql:23]: Do we need this step? IIRC /usr/local/bin is already in the PATH variable

Comment (adivinho) [reply proxysql/build-binary-proxysql]: Yes. This line is redundant. I have removed it.

File: proxysql/build-binary-proxysql [hunk 2]
```diff
@@ -12,7 +12,16 @@ set -o xtrace
...
     RHEL=$(rpm --eval %rhel)
     if [[ $RHEL -eq 8 ]]; then
-        sudo yum -y install cmake python2 gnutls-devel libtool || true
+        sudo yum -y install python2 gnutls-devel libtool || true
+        sudo yum remove -y cmake
+        sudo wget https://github.com/Kitware/CMake/releases/download/v3.19.8/cmake-3.19.8.tar.gz
+        sudo tar -zxvf cmake-3.19.8.tar.gz
+        cd cmake-3.19.8/
+        sudo ./bootstrap
+        sudo make
```
Comment (mohitj1988) [inline proxysql/build-binary-proxysql:21]: Why not club both lines together
sudo make install

File: proxysql/build-binary-proxysql [hunk 3]
```diff
@@ -12,7 +12,16 @@ set -o xtrace
...
+        sudo make install
+        sudo ln -s /usr/local/bin/cmake /usr/bin/cmake
```
Comment (mohitj1988) [inline proxysql/build-binary-proxysql:23]: Is there a specific reason for creating this softlink ?

File: proxysql/build-binary-proxysql [hunk 4]
```diff
@@ -36,6 +45,7 @@ sudo wget https://dl.google.com/go/go1.17.7.linux-amd64.tar.gz
 # Hence using the default $PATH that is available to every new session (both parent and child script)
 sudo tar -C /usr/local/ -xzf go1.17.7.linux-amd64.tar.gz
 sudo cp /usr/local/go/bin/go /usr/local/bin/go
+sudo ln -s /usr/local/go/bin/go /usr/bin/go
```
Comment (mohitj1988) [inline proxysql/build-binary-proxysql:48]: Why are we creating this softlink when we already have go installed at
/usr/local/bin and this path exists in $PATH already

File: proxysql/build-binary-proxysql [hunk 5]
```diff
@@ -12,7 +12,16 @@ set -o xtrace
 if [ -f /usr/bin/yum ]; then
     RHEL=$(rpm --eval %rhel)
     if [[ $RHEL -eq 8 ]]; then
-        sudo yum -y install cmake python2 gnutls-devel libtool || true
+        sudo yum -y install python2 gnutls-devel libtool || true
+        sudo yum remove -y cmake
+        sudo wget https://github.com/Kitware/CMake/releases/download/v3.19.8/cmake-3.19.8.tar.gz
+        sudo tar -zxvf cmake-3.19.8.tar.gz
```
Comment (mohitj1988) [inline proxysql/build-binary-proxysql:18]: Remove the verbose option -v.

File: proxysql/build-binary-proxysql (hunk 4 above)
Comment (adivinho) [reply proxysql/build-binary-proxysql:48]: I have tested on the same docker image that is used for building.
Screenshot 2022-06-16 at 17 58 08

File: proxysql/build-binary-proxysql (hunk 3 above)
Comment (adivinho) [reply proxysql/build-binary-proxysql:23]: Screenshot 2022-06-16 at 18 01 43

File: proxysql/build-binary-proxysql (hunk 2 above)
Comment (adivinho) [reply proxysql/build-binary-proxysql:21]: Such commands sequence are taken from the official cmake site.
Screenshot 2022-06-16 at 18 08 12

File: proxysql/build-binary-proxysql (hunk 3 above)
Comment (mohitj1988) [reply proxysql/build-binary-proxysql:23]: Ah, now I understand why this behaviour is seen. You can retain the softlinks but let me explain the root cause.

Earlier, during the same shell session, cmake existed and that executable was found in /usr/bin, but that was uninstalled.

You then installed another cmake executable in /usr/local/bin.

The bash shell caches the first location that it finds for any external command that you use, which means it doesn't have to do an expensive search for the executable the next time you use the same command. The downside of this is that it won't bother to look again when you, later, install another executable by the same name, even if it's done in a directory that occurs earlier in $PATH than the original location.

The solution to this issue is to empty the cached locations of executables that bash keeps. This is done with 
(hash -r) . To only forget the location of the cmake executable, use hash -d cmake.

File: proxysql/build-binary-proxysql (hunk 4 above)
Comment (mohitj1988) [reply proxysql/build-binary-proxysql:48]: You can retain the softlinks

**Review by mohitj1988 — APPROVED** [review]
Looks good

**Pull request merged**